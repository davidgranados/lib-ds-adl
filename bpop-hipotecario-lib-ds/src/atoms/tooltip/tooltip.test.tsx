import React from "react";
import { mount, ReactWrapper } from "enzyme";
import Tooltip from "./tooltip";

describe("<Tooltip />", () => {
  let wrapper: ReactWrapper<typeof Tooltip>;
  const tooltipTitle = "tooltip title";
  const tooltipBody = "tooltip body";
  beforeEach(() => {
    jest.clearAllMocks();
    // given
    wrapper = mount<typeof Tooltip>(
      <Tooltip textTitle={tooltipTitle} textBody={tooltipBody}>
        Click Me!
      </Tooltip>
    );
  });
  it("should match snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
