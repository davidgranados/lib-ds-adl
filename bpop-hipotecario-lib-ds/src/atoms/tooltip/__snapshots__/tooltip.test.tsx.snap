// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Tooltip /> should match snapshot 1`] = `
<div
  className="wrapper"
  data-testid="tooltip-wrapper"
  id="tooltip-wrapper"
  onClick={[Function]}
>
  <div
    className="content-wrapper content-wrapper--static-width"
    data-testid="tooltip"
    id="tooltip"
  >
    <div
      className="content-body"
    >
      <span
        className="icon-wrapper"
      >
        <span
          className="icon icon-close icon-close"
          data-testid="tooltip-icon-close"
          id="tooltip-icon-close"
        />
      </span>
      <div
        className="content-text"
      >
        <span>
          tooltip title
        </span>
        <span>
          tooltip body
        </span>
      </div>
    </div>
  </div>
  Click Me!
</div>
`;

exports[`<Tooltip /> should match snapshot fullwidth 1`] = `
<div
  className="wrapper"
  data-testid="tooltip-wrapper"
  id="tooltip-wrapper"
  onClick={[Function]}
>
  <div
    className="content-wrapper"
    data-testid="tooltip"
    id="tooltip"
    style={
      Object {
        "width": "calc(calc(100vw - 20px) - 20px)",
      }
    }
  >
    <div
      className="content-body"
    >
      <span
        className="icon-wrapper"
      >
        <span
          className="icon icon-close icon-close"
          data-testid="tooltip-icon-close"
          id="tooltip-icon-close"
        />
      </span>
      <div
        className="content-text"
      >
        <span>
          tooltip title
        </span>
        <span>
          tooltip body
        </span>
      </div>
    </div>
  </div>
  Click Me!
</div>
`;
